chapter_recurrent-modern/seq2seq.md

```diff
@@ -453,11 +527,35 @@ where the last dimension of the tensor stores the predicted token distribution.
 ```{.python .input}
 %%tab all
 decoder = Seq2SeqDecoder(vocab_size, embed_size, num_hiddens, num_layers)
-state = decoder.init_state(encoder(X))
-outputs, state = decoder(X, state)
+if tab.selected('mxnet', 'pytorch', 'tensorflow'):
+    state = decoder.init_state(encoder(X))
+    outputs, state = decoder(X, state)
+if tab.selected('jax'):
+    state = decoder.init_state(encoder.init_with_output(d2l.get_key(), X)[0])
+    (outputs, state), _ = decoder.init_with_output(d2l.get_key(), X, state)
+
 
 d2l.check_shape(outputs, (batch_size, num_steps, vocab_size))
-if tab.selected('mxnet', 'pytorch'):
+if tab.selected('mxnet', 'pytorch', 'jax'):
+    d2l.check_shape(state, (num_layers, batch_size, num_hiddens))
+if tab.selected('tensorflow'):
+    d2l.check_len(state, num_layers)
+    d2l.check_shape(state[0], (batch_size, num_hiddens))
+```
+
+```{.python .input}
+%%tab all
+decoder = Seq2SeqDecoder(vocab_size, embed_size, num_hiddens, num_layers)
+if tab.selected('mxnet', 'pytorch', 'tensorflow'):
+    state = decoder.init_state(encoder(X))
+    outputs, state = decoder(X, state)
+if tab.selected('jax'):
+    state = decoder.init_state(encoder.init_with_output(d2l.get_key(), X)[0])
+    (outputs, state), _ = decoder.init_with_output(d2l.get_key(), X, state)
+
+
+d2l.check_shape(outputs, (batch_size, num_steps, vocab_size))
+if tab.selected('mxnet', 'pytorch', 'jax'):
```

Comment:
redundant?

Reply:
What is redundant?

Reply:
Ohh, i see, somehow a couple of tabs were copied. Turns out the final commit doesn't include the final state of the saved notebook on my local machine. Sorry my bad.